src/VirtualClient/VirtualClient.Dependencies/AMDGPUDriverInstallation.cs

```diff
@@ -107,17 +141,192 @@ await this.stateManager.SaveStateAsync(nameof(AMDGPUDriverInstallation), new Sta
+        {
+            if (string.IsNullOrWhiteSpace(this.LinuxInstallationFile))
+            {
+                throw new DependencyException($"The linux installation file can not be null or empty and it is: {this.LinuxInstallationFile}", ErrorReason.DependencyNotFound);
+            }
+
+            // The .bashrc file is used to define commands that should be run whenever the system
+            // is booted. For the purpose of the AMD GPU driver installation, we want to include extra
+            // paths in the $PATH environment variable post installation.
+            string bashRcPath = $"/home/{this.Username}/.bashrc";
+
+            // We hit a bug where the .bashrc file does not exist on the system. To prevent issues later
+            // we are creating the file if it is missing.
+            if (!this.fileSystem.File.Exists(bashRcPath))
+            {
+                await this.fileSystem.File.WriteAllLinesAsync(
+                    bashRcPath,
+                    new string[]
+                    {
+                        "# ~/.bashrc: executed by bash(1) for non-login shells.",
+                        "# see /usr/share/doc/bash/examples/startup-files (in the package bash-doc)",
+                        "# for examples"
+                    },
+                    cancellationToken);
+            }
+
+            List<string> prerequisiteCommands = this.PrerequisiteCommands(linuxDistribution);
+            List<string> installationCommands = this.VersionSpecificInstallationCommands(linuxDistribution);
+            List<string> postInstallationCommands = this.PostInstallationCommands();
+
+            List<List<string>> commandsLists = new List<List<string>>
+            {
+                prerequisiteCommands,
+                installationCommands,
+                postInstallationCommands
+            };
+
+            foreach (var commandsList in commandsLists)
+            {
+                foreach (string command in commandsList)
+                {
+                    IProcessProxy process = await this.ExecuteCommandAsync(command, null, Environment.CurrentDirectory, telemetryContext, cancellationToken, true)
+                        .ConfigureAwait(false);
+
+                    if (!cancellationToken.IsCancellationRequested)
+                    {
+                        await this.LogProcessDetailsAsync(process, telemetryContext, "AMDGPUDriverInstallation", logToFile: true)
+                            .ConfigureAwait(false);
+
+                        process.ThrowIfErrored<DependencyException>(errorReason: ErrorReason.DependencyInstallationFailed);
+                    }
+                }
+            }
+
+        }
+
+        private List<string> PrerequisiteCommands(LinuxDistribution linuxDistribution)
+        {
+            List<string> commands = new List<string>();
+
+            switch (linuxDistribution)
+            {
+                case LinuxDistribution.Ubuntu:

```

Comment:
It is supported on other distros, but we don't have Linux machine with AMD GPU and other distros, so dont want to add for other distros till we can not test. These driver installations on linux are pretty fragile. Please let me know if you think otherwise